### DISCUSSION TITLE
Alignment of figure captions

### DISCUSSION THREAD

**Discussion by snhansen** [quarto-dev/quarto-cli#7003] [Q&A] [answered]
### Description

I used to be able to center figure captions using CSS in the following way
```
---
format: html
css: styles.css
---

```{r}
#| fig-cap: "Caption."
library(ggplot2)
ggplot(aes(x = wt, y = hp), data = mtcars) +
  geom_point()
```

with `styles.css` containing the following:

```
.figure-caption {
  margin: auto;
  text-align: center;
}
```
I just upgraded to 1.4.383 and this stopped working. This worked in 1.4.309, so I wonder what changed since then? Is there a new method to obtain centered figure captions?

**Comment by cderv** [comment, marked as answer]
#4944 happens and #6620 was merged for dealing with first part of it. 

This changed the way captions are expressed in HTML, so you should just inspect your HTML result, and look at the new rule you should set for your HTML

Probably 
````css
figcaption {
  margin: auto;
  text-align: center;
}
````

or something more targetted. 

See here for the change that happens in HTML : https://quarto.org/docs/prerelease/1.4/crossref.html#changes-in-html-output

Advice: when you are using the pre-release version, watch out for changelog at https://quarto.org/docs/download/prerelease.html
We usually try to document such breaking change like this HTML changes.

**Reply by snhansen** [reply]
Thanks @cderv. I'll keep an eye on the changelog in the future.

The following ended up working (`figcaption` didn't)
```
.figure {
  margin: auto;
  text-align: center;
}
```

**Reply by cderv** [reply]
Glad you made it works. `figcaption` worked in the simple example 
````markdown
---
format: html
---


```{css}
/*| echo: false */
figcaption {
  margin: auto;
  text-align: center;
}
```


```{r}
#| fig-cap: "Caption."
library(ggplot2)
ggplot(aes(x = wt, y = hp), data = mtcars) +
  geom_point()
```
````

But any other rules that applied is good !

**Reply by snhansen** [reply]
Ah yes, I used `.figcaption` instead of `figcaption`. Works here for the example now as well.
